pkg/streamingpromql/testdata/ours/native_histograms.test

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: AGPL-3.0-only
+# Provenance-includes-location: https://github.com/prometheus/prometheus/tree/main/promql/testdata/native_histograms.test
+# Provenance-includes-license: Apache-2.0
+# Provenance-includes-copyright: The Prometheus Authors
+
+# Minimal valid case: an empty histogram.
+load 5m
+	empty_histogram	{{}}
+
+eval instant at 5m empty_histogram
+	{__name__="empty_histogram"} {{}}
+
+# buckets:[1 2 1] means 1 observation in the 1st bucket, 2 observations in the 2nd and 1 observation in the 3rd (total 4).
+load 5m
+	single_histogram	{{schema:0 sum:5 count:4 buckets:[1 2 1]}}
+
+eval instant at 5m single_histogram
```

Comment:
(note: this duplicates some of `pkg/streamingpromql/testdata/upstream/native_histograms.test.disabled` but because `histogram_count` hasn't been implemented yet it differs slightly. This file, or at least these checks, should be redundant once we can support all of the upstream tests).

Reply:
I think we can start using some of `pkg/streamingpromql/testdata/upstream/native_histograms.test.disabled` - there are a few test cases in there that use raw selectors, and then this file can contain extra scenarios the upstream tests don't cover.

I'd like us to get in the habit of using the upstream test cases so we don't miss edge cases (even if in this case, the scenarios are very straightforward).

Eventually we'll remove the `upstream` directory entirely and just use the tests as-is from Prometheus alongside our own test cases.

Reply:
All of the tests in there implement some kind of function such as `histogram_count`, including the tests for `single_histogram`. Once we implement one or more operators then I agree we should remove this file and switch over.

Reply:
>All of the tests in there implement some kind of function such as `histogram_count`, including the tests for `single_histogram`.

There are a small number (eg. [here](https://github.com/grafana/mimir/blob/04e6b7fc0f305f344b0e487691218d898633410a/pkg/streamingpromql/testdata/upstream/native_histograms.test.disabled#L10), [here](https://github.com/grafana/mimir/blob/04e6b7fc0f305f344b0e487691218d898633410a/pkg/streamingpromql/testdata/upstream/native_histograms.test.disabled#L121) and [here](https://github.com/grafana/mimir/blob/04e6b7fc0f305f344b0e487691218d898633410a/pkg/streamingpromql/testdata/upstream/native_histograms.test.disabled#L158)) that don't rely on the `histogram_...` functions. In this case, I'm OK if you just want to copy them here, but in general I'd prefer to use the upstream test cases as-is (even with the pain of commenting out the test cases we don't yet support) to make sure we don't miss anything exercised by the upstream cases.

>Once we implement one or more operators then I agree we should remove this file and switch over.

There will always be a need for this file (unless we upstream the extra test cases we add here) - for example, the upstream test cases only cover instant queries, but we'll at least want some range query test cases. (It's important we test range queries because Prometheus' engine largely treats range queries as a series of instant queries, whereas our engine does things a little differently, so I expect some issues that will only crop up in our engine when running range queries.)

Reply:
Okay, fair enough. I've commented them out and enabled the upstream tests.